cartridge/shop/models.py

```diff
@@ -129,6 +141,8 @@ def save(self, *args, **kwargs):
         if updating and not settings.SHOP_USE_VARIATIONS:
             default = self.variations.get(default=True)
             self.copy_price_fields_to(default)
+        elif hasattr(self, 'content_model'):
+            self.content_model = self._meta.object_name.lower()
```

Comment:
In any case, this should use `ContentTyped.get_content_model_name()`.

I *think* the correct logic is: if the current class is the first non-abstract descendent of `ContentTyped` in its class hierarchy, `content_model` should be `None`.